packages/cli/commands/upload.js

```diff
@@ -61,11 +67,32 @@ exports.handler = async options => {
   const mode = getMode(options);
 
   const uploadPromptAnswers = await uploadPrompt(options);
-
+  const processFieldsOpt = options.processFieldsJs;
   const src = options.src || uploadPromptAnswers.src;
-  const dest = options.dest || uploadPromptAnswers.dest;
+  const saveOutput = options.saveOutput;
+  let dest = options.dest || uploadPromptAnswers.dest;
 
   const absoluteSrcPath = path.resolve(getCwd(), src);
+
+  // The theme.json file must always be at the root of the project - so we look for that and determine the root path based on it.
+  const projectRoot = path.dirname(getThemeJSONPath(absoluteSrcPath));
+  const processFieldsJs =
+    isProcessableFieldsJs(projectRoot, absoluteSrcPath) && processFieldsOpt;
+  let fieldsJs;
```

Comment:
The branching associated with `processFieldsJs` is a somewhat akward. Also I find it strange that were instantiating `FieldsJS` and then mutating the `outputPath` with an object method ... why can't it just do this implicitly?